### ISSUE TITLE
Autocomplete and Linting are Broken

### ISSUE THREAD

**Issue by OutOfFocus4** [redhat-developer/vscode-yaml#508]
Issue Type: <b>Bug</b>

1. With extension version 0.18.0 installed, create a file named `.gitlab-ci.yml`
2. The window looks like this:
![image](https://user-images.githubusercontent.com/50265209/118276771-02342d00-b496-11eb-91c5-8e1e951d9704.png)
*Note the "Gitlab CI configuration (gitlab-ci.json)" above line 1*
3. Autocomplete and linting work fine:
![image](https://user-images.githubusercontent.com/50265209/118276928-390a4300-b496-11eb-8974-36d2d4db0182.png)
4. Update the extension to 0.19.0.
![image](https://user-images.githubusercontent.com/50265209/118277035-57703e80-b496-11eb-8e2d-01eced7c235b.png)
*"Gitlab CI configuration (gitlab-ci.json)" is gone*
5. Linting and autocomplete no longer work
![image](https://user-images.githubusercontent.com/50265209/118277211-8a1a3700-b496-11eb-9ecd-2485de9909e5.png)

Extension version: 0.19.0
VS Code version: Code 1.56.2 (054a9295330880ed74ceaedda236253b4f39a335, 2021-05-12T17:13:13.157Z)
OS version: Windows_NT x64 10.0.19042

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-6700 CPU @ 3.40GHz (8 x 3408)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>skia_renderer: enabled_on<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|31.87GB (27.45GB free)|
|Process Argv|--crash-reporter-id 87f27f2c-dd1b-40a9-ae8b-7ff74ea6d775|
|Screen Reader|no|
|VM|0%|
</details>

**Comment by evidolob** [comment]
@OutOfFocus4 I just test `.gitlab-ci.yml` file and autocomplete works for me on `0.19.0`:
<img width="596" alt="Screenshot 2021-05-14 at 16 31 04" src="https://user-images.githubusercontent.com/929743/118277871-f49c9880-b4d1-11eb-99e3-ed2027424482.png">

Can you check that you have access to https://www.schemastore.org/api/json/catalog.json ?

**Comment by OutOfFocus4** [comment]
I tried on a Mac and it didn't work, either. I can reach the schema store.
![Screen Shot 2021-05-14 at 10 08 42 AM](https://user-images.githubusercontent.com/50265209/118283064-f5ff9e00-b49c-11eb-9b75-e3c9156e16d8.png)

**Comment by OutOfFocus4** [comment]
I did some more investigating and found that if I uninstall the extension and delete the JSON schema cache from my filesystem, the cache will be regenerated if I install version 0.18 of the extension. If I remove the extension, delete the cache, and install version 0.19, the cache directory isn't created.

**Comment by evidolob** [comment]
Hm, that is very strange, we not change anything related to schema cache in `0.19.0`

**Comment by evidolob** [comment]
@OutOfFocus4 I think I know what reason of this issue. 
On `0.19.0` we upgrade to latest json-languageservice(which we use to implement most features) an in latest release they changed way to handle file match patterns. Now it(we) respects glob patterns.
In JSON Schema store `gitlab-ci` defined as:
```json
{
      "name": "gitlab-ci",
      "description": "JSON schema for configuring Gitlab CI",
      "fileMatch": [
        "*.gitlab-ci.yml"
      ],
      "url": "https://json.schemastore.org/gitlab-ci.json"
 }
```
Prior `0.19.0` we always match end of file name(basically we doesn't support glob patterns, even if our docs says opposite).

To test my guess, just rename your file to `<some>.gitlab-ci.yml` to match `fileMatch` in JSON Schema

**Comment by OutOfFocus4** [comment]
I renamed `.gitlab-ci.yml` to `a.gitlab-ci.yml`, and it started to work.

Shouldn't `.gitlab-ci.yml` work with `*.gitlab-ci.yml`, since `*` can match zero characters?

**Comment by evidolob** [comment]
Yes it should, I create PR to fix that https://github.com/redhat-developer/yaml-language-server/pull/467

**Comment by evidolob** [comment]
@OutOfFocus4 Can you test latest `0.19.2`? It supposed to work.

**Comment by OutOfFocus4** [comment]
> @OutOfFocus4 Can you test latest `0.19.2`? It supposed to work.

I updated to `0.19.2`. `.gitlab-ci.yml` still isn't seen as a GitLab CI configuration file.

![image](https://user-images.githubusercontent.com/50265209/119259012-33011a00-bb9a-11eb-867e-10c45f5b06b0.png)

**Comment by evidolob** [comment]
@OutOfFocus4 
I spend several hours to investigate this, and cannot find any clue why it didn't work for you.
On my side it works:

https://user-images.githubusercontent.com/929743/119780796-e9366d80-bed2-11eb-98d9-98b69d0ad641.mov


The only possible reason it that your PC cannot load schema from jsonschema store. 
Can you make sure that you can get https://www.schemastore.org/api/json/catalog.json and https://json.schemastore.org/gitlab-ci.json

**Comment by OutOfFocus4** [comment]
I just tried again on Windows, MacOS, and Ubuntu.

Windows 10:
![image](https://user-images.githubusercontent.com/50265209/119829610-6957d580-bec9-11eb-9e70-f729c5b5e643.png)

MacOS:
![YAML MacOS](https://user-images.githubusercontent.com/50265209/119829633-707ee380-bec9-11eb-8c8f-31b380ab8e50.png)

Ubuntu:
![YAML Ubuntu](https://user-images.githubusercontent.com/50265209/119829646-75439780-bec9-11eb-9def-6d5b5d864d21.png)

Autocomplete and linting work in `a.gitlab-ci.yml`, but not in `.gitlab-ci.yml` with identical contents in the same directory.

**Comment by JPinkney** [comment]
I can actually reproduce this on my end. It looks like the glob pattern from the schema store isn't working. I can also reproduce this without the schema store by doing:
```
"yaml.schemas": {
        "https://json.schemastore.org/gitlab-ci.json": "*.gitlab-ci.yml"
},
```
on 0.19.2

**Comment by evidolob** [comment]
@JPinkney Thx, it seems that I need to start investigation from scratch

**Comment by evidolob** [comment]
@JPinkney @OutOfFocus4 
I give up, I try several laptops (with macos, windows and linux) and not able to reproduce this, but all of them used same wify/router to internet access.
The only reason for this I see in yours internet connection, that `https://json.schemastore.org/gitlab-ci.json` is not available for some reasons.

**Comment by OutOfFocus4** [comment]
As I demonstrated in my screenshots in [my screenshots above](https://github.com/redhat-developer/vscode-yaml/issues/508#issuecomment-849615238), the extensions works with files named `a.gitlab-ci.yml`, but not `.gitlab-ci.yml`.

Is the globbing using an OS-level search function, and my OS preferences make it ignore dotfiles while yours includes dotfiles?

**Comment by OutOfFocus4** [comment]
I just updated to 0.20.0, and everything is working again. Should I close this issue, or keep it open until a root cause has been identified and regression tests are added?

**Comment by evidolob** [comment]
Let's wait a bit. 
That strange we do not change anything on schema matching between `0.19.2` and `0.20.0` .

**Comment by fiffly** [comment]
Autocomplete does not work on Windows 10 and version v0.22.0. I tried on an other computer (Windows 11), it doesn't work more.
Downgrading does not solve problem
